combine_dimensions_D.m

```diff
@@ -48,6 +48,8 @@
     tmp = A * B;
     
     Index = deg^num_dims*(i-1)+1:deg^num_dims*i;
+    % Captain! different parts of fval set on each iteration with no overlap. The same time
```

Comment:
I don't understand the "different parts of fval" comment. Are you suggesting an optimization?

As for the time multiplier, we'd discussed this some, and there may be cases where each spatial dimension function has a different time multiplier, hence the way it is done here.

Reply:
This comment was an annotation meant for me to help conceptualize what was happening - not something that would be present in the final version. I believe I was making note for myself that loop iterations did not carry data dependencies into later loop iterations.